pkg/apis/certificates/v1alpha1/types.go

```diff
@@ -46,6 +46,12 @@ type CertificateSigningRequestSpec struct {
 	// Base64-encoded PKCS#10 CSR data
 	Request []byte `json:"request" protobuf:"bytes,1,opt,name=request"`
 
+	// allowedUsages specifies a set of usage context the key will be
+	// valid for.
+	// See: https://tools.ietf.org/html/rfc5280#section-4.2.1.3
+	//      https://tools.ietf.org/html/rfc5280#section-4.2.1.12
+	AllowedUsages []KeyUsage `json:"keyUsage,omitempty" protobuf:"bytes,5,opt,name=keyUsage"`
```

Comment:
"AllowedUsages" feels weird in a spec. Would "RequestedUsages" or just "Usages" be better?